controller/controller_server/exception_handler.py

```diff
@@ -0,0 +1,42 @@
+from decorator import decorator
+
+import grpc
+
+import controller.array_action.errors as controller_errors
+from controller.common.csi_logger import get_stdout_logger
+from controller.controller_server.errors import ValidationException
+
+logger = get_stdout_logger()
+
+status_codes_by_exception = {
+    ValidationException: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.IllegalObjectName: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.SpaceEfficiencyNotSupported: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.PoolDoesNotExist: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.PoolDoesNotMatchCapabilities: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.PoolParameterIsMissing: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.ExpectedSnapshotButFoundVolumeError: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.ObjectNotFoundError: grpc.StatusCode.NOT_FOUND,
+    controller_errors.PermissionDeniedError: grpc.StatusCode.PERMISSION_DENIED,
+    controller_errors.VolumeAlreadyExists: grpc.StatusCode.ALREADY_EXISTS,
+    controller_errors.NotEnoughSpaceInPool: grpc.StatusCode.RESOURCE_EXHAUSTED
+}
+
+
+def handle_exception(response_type, context, ex):
+    logger.exception(ex)
+    context.set_details(str(ex))
+    status_code = status_codes_by_exception.get(type(ex), grpc.StatusCode.INTERNAL)
```

Comment:
I think this should come from a parameter

Reply:
I don't like lots of parameters in functions, but I can't find something better than that at the moment. Done